src/wallet/spend.cpp

```diff
@@ -711,8 +728,18 @@ util::Result<SelectionResult> AutomaticCoinSelection(const CWallet& wallet, Coin
                 if (HasErrorMsg(res)) res_detailed_errors.emplace_back(res);
             }
         }
-        // Coin Selection failed.
-        return res_detailed_errors.empty() ? util::Result<SelectionResult>(util::Error()) : res_detailed_errors.front();
+
+        // Return right away if we have a detailed error
+        if (!res_detailed_errors.empty()) return res_detailed_errors.front();
+
+        // Let's check if we have long-chain of unconfirmed UTXOs in the not accepted groups
+        for (const auto& group : not_accepted_groups) {
+            if (group.m_ancestors >= max_ancestors)
+                return util::Result<SelectionResult>({_("Unconfirmed UTXOs are available, but spending them creates a chain of transactions that will be rejected by the mempool")});
```

Comment:
Just because these UTXOs are available it doesn't mean that including them would cover required amount.

For example let's imaging you have two coins in a wallet:
- 48 BTC with 6 confirmation
- 1 BTC with long unconfirmed ancestry

If you try to send 50 BTC you should get "insufficient funds" error, but IIUC after this PR you'll get "Unconfirmed UTXOs are available..." which is unrelated.

Reply:
yeah good, fixed.
also, pushed an early return if the sum of the not accepted groups decreases the available amount below the target (we simply don't have enough balance after applying the filters, so no need to run the selection algos).